src/uwtools/set_config.py

```diff
@@ -105,6 +114,16 @@ def create_config_obj(argv):
 
         config_obj.update_values(user_config_obj)
 
+    if user_args.dry_run:
+        if user_args.outfile:
+            out_object = user_args.outfile
+            log.info(f'warning file {out_object} ',
+                 r"not written when using --dry_run")
+        # apply switch to allow user to view the results of config
+        # instead of writing to disk
+    if config_obj is not None:
+        log.info(config_obj)
```

Comment:
With a return above, you can likely remove the if-statement and just `log.info(config_obj)`, then `return`. For the dry-run portion, we don't even need to go through the next bit of code.